dids/src/main/kotlin/web5/sdk/dids/did/BearerDid.kt

```
@@ -0,0 +1,143 @@
+   *
+   * This function takes a Verification Method selector that can be used to select a specific verification method
+   * from the DID Document if desired. If no selector is provided, the payload will be signed with the key associated
+   * to the first verification method in the DID Document.
+   *
+   * The selector can either be a Verification Method ID or a Purpose. If a Purpose is provided, the first verification
+   * method in the DID Document that has the provided purpose will be used to sign the payload.
+   *
+   * The returned signer is a function that takes a byte payload and returns a byte signature.
+   */
+  @JvmOverloads
+  public fun getSigner(selector: VMSelector? = null): Pair<DidSigner, VerificationMethod> {
+    val verificationMethod = document.selectVerificationMethod(selector)
+
+    val keyAliasResult = runCatching { verificationMethod.publicKeyJwk?.computeThumbprint() }
+    val keyAlias = keyAliasResult.getOrNull() ?: throw Exception("Failed to compute key alias")
+
+    val signer: DidSigner = { payload ->
+      keyManager.sign(keyAlias, payload)
+    }
+
+    return Pair(signer, verificationMethod)
+  }
+
+  /**
+   * Converts a `BearerDid` object to a portable format containing the URI and verification methods
+   * associated with the DID.
+   *
+   * This method is useful when you need to represent the key material and metadata associated with
+   * a DID in format that can be used independently of the specific DID method implementation. It
+   * extracts both public and private keys from the DID's key manager and organizes them into a
+   * `PortableDid` structure.
+   *
+   * @returns A `PortableDid` containing the URI, DID document, metadata, and optionally private
+   *          keys associated with the `BearerDid`.
+   */
+  public fun export(): PortableDid {
+
+    val keyExporter = keyManager as? KeyExporter
```

we should do an `is KeyExporter` check and only run the exports if the keymanager is an exporter.  otherwise ignore and proceed

---

by `ignore and proceed`, do you mean we should throw an exception, or something else? like have this check at the top

```kotlin
check(keyManager is KeyExporter) {
      "KeyManager must implement KeyExporter to export keys"
    }
```